data_creation/data_creation.py

```diff
@@ -0,0 +1,357 @@
+
+        valid_gloss_count = 0
+        replaced_gloss_count = 0
+
+        for gloss, start, end in zip(glosses, gloss_start_times, gloss_end_times):
+            gloss = gloss.decode('utf-8')
+            start_frame = math.floor(start / 1000 * fps)
+            end_frame = math.ceil(end / 1000 * fps)
+
+            # Adjust gloss frame range relative to the sentence's start frame
+            relative_start_frame = start_frame - sentence_start_frame
+            relative_end_frame = end_frame - sentence_start_frame
+
+            print(f"Processing sentence id: {sentence_id}")
+            print(f"Processing gloss: {gloss}")
+            # print(f"absolute_start_frame: {start_frame}, absolute_end_frame: {end_frame}, data_size: {len(pose_body.data)}")
+            print(f"relative_start_frame: {relative_start_frame}, relative_end_frame: {relative_end_frame}")
+
+            # Ensure relative frame range is within the sentence's local frame range
+            if relative_start_frame < 0 or relative_end_frame > len(pose_body.data):
+                print(f"Skipping Gloss '{gloss}' due to frame range out of bounds.")
+                continue
+
+            valid_gloss_count += 1
+
+
+            # Extract original pose for current gloss
+            original_pose_body = pose_body.select_frames(range(relative_start_frame, relative_end_frame))
+            original_pose_header = self._get_pose_header("dgs_corpus")
+            original_pose = Pose(original_pose_header, original_pose_body)
+            original_poses_sequence.append(original_pose)
+
+
+            # Determine if the gloss should be replaced
+            if self._should_replace(gloss):
+                print(f"Original pose shape: {original_pose_body.data.shape}")
+                replaced_gloss_count += 1
+                types_gloss_pose_path = self.dictionary[gloss]['views']['pose']
+                with open(types_gloss_pose_path, "rb") as f:
+                    types_gloss_pose = Pose.read(f.read())
+
+                    # original_frame_count = len(original_pose_body.data)
+                    # current_frame_count = len(types_gloss_pose.body.data)
+                    # print(f"Original frame count: {original_frame_count}")
+                    # print(f"Current frame count: {current_frame_count}")
+                    print(f"Updated pose shape before interpolation: {types_gloss_pose.body.data.shape}")
+
+                    # Interpolate pose if fps of dgs types pose is different from dgs corpus pose
+                    if types_gloss_pose.body.fps != original_pose_body.fps:
+                        interpolated_pose_body = types_gloss_pose.body.interpolate(new_fps=original_pose_body.fps)
+                        updated_pose_header = self._get_pose_header("dgs_types")
+                        updated_pose = Pose(updated_pose_header, interpolated_pose_body)
+                        print(f"Updated pose shape after interpolation: {interpolated_pose_body.data.shape}")
```

Comment:
my recommendation for here is that you don't deal with interpolations. you just:
 `assert concatenated_updated_pose.body.fps == concatenated_original_pose.body.fps`

then, to make it actually happen, when loading the `dgs_corpus`, you can specify `fps=25` and that would be done for you.